flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/hybrid/tiered/shuffle/TieredInternalShuffleMaster.java

```diff
@@ -0,0 +1,85 @@
+import org.apache.flink.runtime.io.network.partition.hybrid.tiered.common.TieredStorageConfiguration;
+import org.apache.flink.runtime.io.network.partition.hybrid.tiered.storage.ResourceRegistry;
+import org.apache.flink.runtime.io.network.partition.hybrid.tiered.storage.TieredStorageMasterClient;
+import org.apache.flink.runtime.io.network.partition.hybrid.tiered.tier.TierFactory;
+import org.apache.flink.runtime.io.network.partition.hybrid.tiered.tier.TierMasterAgent;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.apache.flink.runtime.io.network.partition.hybrid.tiered.common.TieredStorageIdMappingUtils.convertId;
+
+/**
+ * A wrapper internal shuffle master class for tiered storage. All the tiered storage operations
+ * with the shuffle master should be wrapped in this class.
+ */
+public class TieredInternalShuffleMaster {
+
+    private final TieredStorageMasterClient tieredStorageMasterClient;
+
+    private final Map<JobID, List<ResultPartitionID>> jobPartitionIds;
+
+    public TieredInternalShuffleMaster(Configuration conf) {
+        TieredStorageConfiguration tieredStorageConfiguration =
+                createTieredStorageConfiguration(conf);
+        ResourceRegistry resourceRegistry = new ResourceRegistry();
+        List<TierMasterAgent> tierFactories =
+                tieredStorageConfiguration.getTierFactories().stream()
+                        .map(TierFactory::createMasterAgent)
+                        .peek(tierMasterAgent -> tierMasterAgent.setup(resourceRegistry))
+                        .collect(Collectors.toList());
+        this.tieredStorageMasterClient = new TieredStorageMasterClient(tierFactories);
+        this.jobPartitionIds = new HashMap<>();
+    }
+
+    public void addPartition(JobID jobID, ResultPartitionID resultPartitionID) {
+        jobPartitionIds.computeIfAbsent(jobID, ignore -> new ArrayList<>()).add(resultPartitionID);
+        tieredStorageMasterClient.addPartition(convertId(resultPartitionID));
+    }
+
+    public void releasePartition(ResultPartitionID resultPartitionID) {
+        tieredStorageMasterClient.releasePartition(convertId(resultPartitionID));
+    }
+
+    public void unregisterJob(JobID jobID) {
+        List<ResultPartitionID> resultPartitionIDs = jobPartitionIds.remove(jobID);
+        if (resultPartitionIDs != null) {
+            resultPartitionIDs.forEach(
+                    resultPartitionID ->
+                            tieredStorageMasterClient.releasePartition(
+                                    convertId(resultPartitionID)));
+        }
+    }
+
+    private TieredStorageConfiguration createTieredStorageConfiguration(Configuration conf) {
+        // TODO, from the configuration, get the configured options(i.e., remote storage path, the
+        // reserved storage size, etc.), then set them to the builder.
```

Comment:
From this description, I think this method should not belong to `TieredInternalShuffleMaster`. Take remote storage path as an example, we need it not only for the master but also for the producer & consumer, and we definitely want to derive the same path from the configuration for all the components. Same for reserved storage size and likely other configurations.

Reply:
Replaced this with a `TieredStorageConfiguration#fromConfiguration`.